mycroft/configuration/config.py

```diff
@@ -183,30 +184,72 @@ def get(configs=None, cache=True):
+                              be loaded
+
 
         Returns:
             (dict) configuration dictionary.
         """
         if Configuration.__config:
             return Configuration.__config
         else:
-            return Configuration.load_config_stack(configs, cache)
+            return Configuration.load_config_stack(configs, cache, remote)
 
     @staticmethod
-    def load_config_stack(configs=None, cache=False):
+    def load_config_stack(configs=None, cache=False, remote=True):
         """Load a stack of config dicts into a single dict
 
         Args:
             configs (list): list of dicts to load
             cache (boolean): True if result should be cached
-
+            remote (boolean): False if the Mycroft Home settings shouldn't
+                              be loaded
         Returns:
             (dict) merged dict of all configuration files
         """
         if not configs:
-            configs = [LocalConf(DEFAULT_CONFIG), RemoteConf(),
-                       LocalConf(SYSTEM_CONFIG), LocalConf(USER_CONFIG),
-                       Configuration.__patch]
+            configs = configs or []
+
+            # First use the patched config
+            configs.append(Configuration.__patch)
+
+            # Then use XDG config
+            # This includes both the user config and
+            # /etc/xdg/mycroft/mycroft.conf
+            for dir in BaseDirectory.load_config_paths('mycroft'):
+                configs.append(LocalConf(join(dir, 'mycroft.conf')))
+
+            # Then check the old user config
+            if isfile(OLD_USER_CONFIG):
+                LOG.warning(" ===============================================")
+                LOG.warning(" ==             DEPRECATION WARNING           ==")
+                LOG.warning(" ===============================================")
+                LOG.warning(" You still have a config file at " +
+                            OLD_USER_CONFIG)
+                LOG.warning(" Note that this location is deprecated and will" +
+                            " not be used in the future")
+                LOG.warning(" Please move it to " +
+                            BaseDirectory.save_config_path('mycroft'))
+                configs.append(LocalConf(OLD_USER_CONFIG))
+
+            # Then check the XDG user config
+            if isfile(USER_CONFIG):
+                configs.append(LocalConf(USER_CONFIG))
+
+            # Then use remote config
+            if remote:
+                configs.append(RemoteConf())
+
+            # Then use the system config (/etc/mycroft/mycroft.conf)
+            configs.append(LocalConf(SYSTEM_CONFIG))
+
+            # Then use the config that comes with the package
+            configs.append(LocalConf(DEFAULT_CONFIG))
+
+            # Make sure we reverse the array, as merge_dict will put every new
+            # file on top of the previous one
+            configs = reversed(configs)
```

Comment:
I originally wondered why each config gets appended and then reversed (as opposed to appending in the correct order to begin with), but presumably this is to handle the order of `load_config_paths()`?

Reply:
It is due to the order returned by the `load_config_paths()` generator as far as I've understood it.

Reply:
`load_config_paths()`:

> Returns an iterator which gives each directory named ‘resource’ in the configuration search path. Information provided by earlier directories should take precedence over later ones, and the user-specific config dir comes first.

So yes, because of that. The most important directories come first, the less important ones later.

Reply:
comment below with suggested refactor into smaller functions addresses this